ingestion-edge/ingestion_edge/publish.py

```diff
@@ -3,13 +3,40 @@
 # file, you can obtain one at http://mozilla.org/MPL/2.0/.
 
 from .conf import ROUTE_TABLE
+from .conf import QUEUE_PATH
 from datetime import datetime
 from flask import Blueprint, request
 from google.cloud.pubsub import PublisherClient
+from persistqueue import Queue
+from persistqueue.exceptions import Empty
 from typing import Tuple
+import threading
 
 publish = Blueprint('publish', __name__)
 client = PublisherClient()
+q = Queue(QUEUE_PATH)
+flush_lock = threading.Lock()
+
+
+@publish.route("/__flush__")
+def flush_queue() -> Tuple[str, int]:
+    if q.qsize() == 0:
+        # early return on empty queue
+        return "", 204
+    tasks_done = 0
+    with flush_lock:
+        while True:
+            try:
+                item = q.get_nowait()
+            except Empty:
+                break
+            try:
+                client.publish(**item).result()
+                q.task_done()
+                tasks_done += 1
+            except Exception:
+                return str(tasks_done), 202
+    return str(tasks_done), 200
```

Comment:
Let's make these 200 and 202 literals named constants to better document intention.